### PR TITLE
Faster patches grid

### PR DESCRIPTION
**Pull request by vovaf709**

### CODE REVIEW

File: dpipe/predict/shape.py [hunk 1]
```diff
@@ -111,8 +112,17 @@ def wrapper(x, *args, **kwargs):
             elif ((shape - local_size) < 0).any() or ((local_stride - shape + local_size) % local_stride).any():
                 raise ValueError('Input cannot be patched without remainder.')
 
-            if stream:
-                patches = predict(divide(x, local_size, local_stride, input_axis, get_boxes=get_boxes), *args, **kwargs)
+            if async_predict:
+                divide_wrapper = make_batch if batch_size is not None else lambda x, batch_size: x
```
Comment (vovaf709) [inline dpipe/predict/shape.py:116]: Don't forget to adapt branch with `async_predict == False` for `batch_size is not None`

File: tests/predict/test_shape.py [hunk 2]
```diff
@@ -16,13 +16,19 @@ def batch_size(request):
...
     return request.param
 
 
-def test_patches_grid(stream):
+@pytest.fixture(params=[False, True])
+def async_predict(request):
+    return request.param
+
+
+def test_patches_grid(use_torch, async_predict):
```
Comment (vovaf709) [inline tests/predict/test_shape.py:28]: Let's also test `batch_size`

File: dpipe/predict/shape.py (hunk 1 above)
Comment (AnihilatorGun) [reply dpipe/predict/shape.py:116]: Done

File: tests/predict/test_shape.py (hunk 2 above)
Comment (AnihilatorGun) [reply tests/predict/test_shape.py:28]: Done

**Review by maxme1 — APPROVED** [review]

**Pull request merged**